### DISCUSSION TITLE
Failed to send chain to the execution channel

### DISCUSSION THREAD

**Discussion by lakosm** [cybertec-postgresql/pg_timetable#427] [Q&A] [answered]
Hi,

I have started multiple (20) jobs using timetable.add_job. These are long running, CPU intensive jobs.
I have checked the timetable.log regularly and I found sometimes the following error messages:
Failed to send chain to the execution channel

But the chain is runing forward.
Could you explain to me, what this message is? 

Thanks!

Regards,
Miklos

**Comment by pashagolub** [comment, marked as answer]
Hello, assuming you're running the latest version, you're out of the capacity of the channel for passing chains to workers.

Channel accumulates jobs and then workers pull one by one job from this channel. If all of your workers are busy and new jobs are fetched for execution channel will be out of capacity to hold new jobs.

Solution:
- increase the number of workers
- make chains less aggressive, e.g. execute not every minute but every 10 minutes
- use `AFTER` interval chains, so new chains will start only AFTER the period of time the last one finished

**Comment by lakosm** [comment]
thank you
